cmd/etcd-launcher/main.go

```diff
@@ -73,10 +78,28 @@ func main() {
 	logOpts := kubermaticlog.NewDefaultOptions()
 	rawLog := kubermaticlog.New(logOpts.Debug, logOpts.Format)
 	log := rawLog.Sugar()
-	log = log.With("member-name", e.config.podName)
 
+	// here we find the cluster state
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		log.Fatalw("failed to get in cluster config", zap.Error(err))
+	}
+	clusterClient, err := ctrlruntimeclient.New(config, ctrlruntimeclient.Options{})
+	if err != nil {
+		log.Fatalw("failed to create cluster client", zap.Error(err))
+	}
+
+	k8cCluster := kubermaticv1.Cluster{}
+	if err := clusterClient.Get(context.Background(), types.NamespacedName{Name: strings.ReplaceAll(e.config.namespace, "cluster-", ""), Namespace: ""}, &k8cCluster); err != nil {
+		log.Fatalw("failed to get cluster", zap.Error(err))
+	}
 	initialMembers := initialMemberList(e.config.clusterSize, e.config.namespace)
 
+	e.config.initialState = "new"
+	if isInitialized := k8cCluster.Status.HasConditionValue(kubermaticv1.ClusterConditionEtcdClusterInitialized, corev1.ConditionTrue); isInitialized {
```

Comment:
isInitialized not used anywhere; if-expression can be reduced to just the function call. Or is this intentional style?

Reply:
seemed more readable this way. Will put a comment instead.